File: data-jpa/src/main/java/top/bettercode/summer/data/jpa/support/Size.java
package top.bettercode.summer.data.jpa.support;

import org.springframework.data.domain.Sort;

/**
 * @author Peter Wu
 */
public class Size {
  private final int size;
  private final Sort sort;

  public static Size of(int size) {
    return of(size, Sort.unsorted());
  }

  public static Size of(int size, Sort sort) {
    return new Size(size, sort);
  }

  private Size(int size, Sort sort) {
    this.size = size;
    this.sort = sort;
  }

  public int getSize() {
    return size;
  }

  public Sort getSort() {
    return sort;
  }
}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/support/Size.kt
package top.bettercode.summer.data.jpa.support

import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort

/**
 * @author Peter Wu
 */
open class Size private constructor(val size: Int, val sort: Sort) {

    fun isUnlimited(): Boolean {
        return size == -1
    }

    companion object {

        private val UNLIMITED: Size = Size(-1, sort = Sort.unsorted())

        @JvmStatic
        fun unlimited(): Size {
            return UNLIMITED
        }

        @JvmStatic
        fun of(pageable: Pageable): Size {
            return of(pageable.pageSize, pageable.sort)
        }

        @JvmStatic
        @JvmOverloads
        fun of(size: Int, sort: Sort = Sort.unsorted()): Size {
            return Size(size, sort)
        }
    }
}
